multicodec-packed.md

```diff
@@ -0,0 +1,111 @@
+# multicodec-packed
+
+[![](https://img.shields.io/badge/made%20by-Protocol%20Labs-blue.svg?style=flat-square)](http://ipn.io)
+[![](https://img.shields.io/badge/project-multiformats-blue.svg?style=flat-square)](http://github.com/multiformats/multiformats)
+[![](https://img.shields.io/badge/freenode-%23ipfs-blue.svg?style=flat-square)](http://webchat.freenode.net/?channels=%23ipfs)
+
+> compact self-describing codecs
+
+## Table of Contents
+
+- [Motivation](#motivation)
+- [How does it work? - Protocol Description](#how-does-it-work---protocol-description)
+- [Multicodec-Packed Protocol Tables](#multicodec-packed-protocol-tables)
+  - [Standard mcp protocol table](#standard-mcp-protocol-table)
+- [Implementations](#implementations)
+- [FAQ](#faq)
+- [Maintainers](#maintainers)
+- [Contribute](#contribute)
+- [License](#license)
+
+## Motivation
+
+[Multicodecs](./README.md) are self-describing protocol/encoding streams. Multicodec-packed is a different representation of multicodec, which uses an agreed-upon "protocol path table". It is designed for use in short strings, such as keys or identifiers (such as [CID](https://github.com/ipld/cid)).
+
+## How does it work? - Protocol Description
+
+`multicodec-packed` is a _self-describing multiformat_, it wraps other formats with a tiny bit of self-description:
+
+```sh
+<mcp-code><encoded-data>
+# or
+<multicodec-packed-varint><encoded-data>
+```
+
+It can also be used as part of identifiers or keys to other data:
+
+```
+# suppose we have a value and a key to retrieve it
+"<key>" -> <value>
+
+# we can use multicodec-packed with the key to know what codec the value is in
+"<mcp><key>" -> <value>
+```
+
+It is worth noting that multicodec-packed works very well in conjunction with [multihash](https://github.com/multiformats/multihash) and [multiaddr](https://github.com/multiformats/multiaddr), as you can prefix those values with a multicodec-packed to tell what they are.
+
+## Multicodec-Packed Protocol Tables
+
+Multicodec-packed uses "protocol tables" to agree upon the mapping from one multicodec-packed code (a single varint). These tables map an `<mcp-code>` to a full [multicodec protocol path](./README.md#the-protocol-path). These tables can be application specific, though -- like [with](https://github.com/multiformats/multihash) [other](https://github.com/multiformats/multibase) [multiformats](https://github.com/multiformats/multiaddr) -- we will keep a globally agreed upon table with common protocols and formats.
+
+### Standard mcp protocol table
+
+This is the standard multicodec-packed protocol table.
+
+#### WARNING: WIP. this table is not ready for wide use.
+
+TODO: see if IANA has a ready-made table for us to use here. Even just a listing of the most popular formats would be good enough.
+
+```sh
+code  codec
+0x00  raw binary data
+0x40  multicodec
```

Comment:
This is self referencing.  Is the idea to use this when a packed-multicodec does not describe the data?